### PR TITLE
improve warning message when entity misaligned

### PR DESCRIPTION
**Pull request by howl-anderson**
**Proposed changes**:
- Improve the warning message when entity misaligned

fixs #7409

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @b-quachtran will take a look at it as soon as possible ✨

File: rasa/nlu/extractors/extractor.py [hunk 1]
```diff
@@ -407,12 +407,26 @@ def check_correct_entity_annotations(training_data: TrainingData) -> None:
...
+                        for t in example.get(TOKENS_NAMES[TEXT])
+                    ]
                     rasa.shared.utils.io.raise_warning(
                         f"Misaligned entity annotation in message '{example.get(TEXT)}' "
                         f"with intent '{example.get(INTENT)}'. Make sure the start and "
-                        f"end values of entities in the training data match the token "
-                        f"boundaries (e.g. entities don't include trailing whitespaces "
-                        f"or punctuation).",
+                        f"end values of entities ('{entities_repr}') in the training "
+                        f"data match the token boundaries ('{tokens_repr}'). "
```
Comment (erohmensing) [inline rasa/nlu/extractors/extractor.py:426]:
```suggestion
                        f"end values of entities ({entities_repr}) in the training "
                        f"data match the token boundaries ({tokens_repr}). "
```

File: rasa/nlu/extractors/extractor.py [hunk 2]
```diff
@@ -407,12 +407,26 @@ def check_correct_entity_annotations(training_data: TrainingData) -> None:
...
+                        "Common causes: 1) entities include trailing whitespaces or punctuation; "
```
Comment (erohmensing) [inline rasa/nlu/extractors/extractor.py:427]:
```suggestion
                        "Common causes: \n  1) entities include trailing whitespaces or punctuation "
```

File: rasa/nlu/extractors/extractor.py [hunk 3]
```diff
@@ -407,12 +407,26 @@ def check_correct_entity_annotations(training_data: TrainingData) -> None:
...
+                        "2) the tokenizer gives an unexpected result, usually happens to "
+                        "languages such as Chinese that don't use whitespace as word separator.",
```
Comment (erohmensing) [inline rasa/nlu/extractors/extractor.py:429]:
```suggestion
                        "\n  2) the tokenizer gives an unexpected result, due to "
                        "languages such as Chinese that don't use whitespace for word separation",
```

**Review by erohmensing — COMMENTED** [review]
Thanks for the addition @howl-anderson. Can we flesh out "the tokenizer gives an unexpected result in a short way? Im presuming its about trying to annotate characters where you expect token boundaries (like trying to annotate `bcd` in `[a][bc][d][e]`) but for some reason the tokenizer doesn't agree (annotating `bcd` when the tokenizer outputs `[a][bc][de]` incorrectly)?

**Comment by howl-anderson** [comment]
> Im presuming its about trying to annotate characters where you expect token boundaries (like trying to annotate `bcd` in `[a][bc][d][e]`) but for some reason the tokenizer doesn't agree (annotating `bcd` when the tokenizer outputs `[a][bc][de]` incorrectly)?

Yes, this is a common warning for Chinese (4 of 196 training examples got this warning, consistent with the estimated value in #6753) if we use a Chinese tokenizer.

**Comment by howl-anderson** [comment]
Here is a sample message this PR currently provides:

```text
UserWarning: Misaligned entity annotation in message '下个星期五在南京' with intent 'weather'. Make sure the start and end values of entities ([(0, 5, '下个星期五'), (6, 8, '南京')]) in the training data match the token boundaries ([(0, 4, '下个星期'), (4, 6, '五在'), (6, 8, '南京')]). Common causes: 
  1) entities include trailing whitespaces or punctuation
  2) the tokenizer gives an unexpected result, due to languages such as Chinese that don't use whitespace for word separation
```

**Review by erohmensing — APPROVED** [review]
Ok, great! as long as you think people who commonly run into this issue know that's what the log is telling them, then I think we're good to go.

**Pull request merged**